task-ballerina/src/task/listener.bal

```diff
@@ -23,19 +23,40 @@ public class Listener {
     boolean started = false;
 
     private TimerConfiguration|AppointmentConfiguration listenerConfiguration;
+    private MisfireConfiguration misfireConfiguration;
 
     # Initializes the `task:Listener` object. This may panic if the initialization is failed due to a configuration
     # error.
     #
     # + configuration - The `task:TimerConfiguration` or `task:AppointmentConfiguration` record to define the
-    #   `task:Listener` behavior
-    public isolated function init(TimerConfiguration|AppointmentConfiguration configuration) {
+    #                   `task:Listener` behavior
+    # + misfireConfiguration - The `task:MisfireConfiguration` record, which is used to configure the misfire situations
+    #                          of the scheduler
+    public isolated function init(TimerConfiguration|AppointmentConfiguration configuration,
+                                  MisfireConfiguration misfireConfiguration = {}) {
+        var noOfRecurrences = configuration["noOfRecurrences"];
```

Comment:
Please check every place for where indexing is used instead of field access. Even if we have to use indexing, shall we define a string constant for the field name and use it, instead of just using like this?

```suggestion
        var noOfRecurrences = configuration.noOfRecurrences;
```